test/metabase/notification/payload/impl/card_test.clj

```diff
@@ -408,14 +408,15 @@
 (deftest send-once-archive-on-first-successful-send
   (notification.tu/with-card-notification
     [notification {:notification-card {:send_once true}}]
-    (testing "do not archive if the send fail for any reason"
-      (mt/with-dynamic-fn-redefs [notification.payload/notification-payload (fn [& _args] (throw (ex-info "error" {})))]
-        (u/ignore-exceptions (notification/send-notification! notification))
-        (is (true? (t2/select-one-fn :active :model/Notification (:id notification))))))
-
-    (testing "archive if the send is successful"
-      (notification/send-notification! notification)
-      (is (false? (t2/select-one-fn :active :model/Notification (:id notification)))))))
+    (let [notification (t2/select-one :model/Notification (:id notification))]
```

Comment:
this test was not catching the bug before because notification here is already fully hydrated, and when sending notification we skip hydration for it.

In the sending notification from quartz path, the notification is not hydrated.